controllers/vrg_volrep.go

```diff
@@ -1466,6 +1466,18 @@ func setPVCClusterDataProtectedCondition(protectedPVC *ramendrv1alpha1.Protected
 	}
 }
 
+// ensureVRDeletedFromAPIServer adds an additional step to ensure that we wait for volumereplication deletion
+// from API server before moving ahead vrg finalizer removal.
+func (v *VRGInstance) ensureVRDeletedFromAPIServer(vrNamespacedName types.NamespacedName) error {
+	if err := v.reconciler.APIReader.Get(v.ctx, vrNamespacedName, v.instance); err != nil {
+		v.log.Info("Wait to delete VolumeReplication resource from API server", "Error", err)
+
+		return err
```

Comment:
We need to return an error stating, resource is still not deleted from the API server or some such, e.g "waiting for VolumeReplication instance (%s/%s) to be deleted"